lib/ansible/galaxy/collection/concrete_artifact_manager.py

```diff
@@ -154,7 +156,7 @@ def get_galaxy_artifact_path(self, collection):
                 validate_certs=self._validate_certs,
                 token=token,
             )  # type: bytes
-        except URLError as err:
+        except Exception as err:
```

Comment:
I kinda hate this wholesale catch. @sivel how about having a separate catch-all except block with a distinct error message being reraised?

Reply:
Ah.. I'm not asking you to give that up right now. I'm asking for an additional except block so that there's two blocks that raise `AnsibleError` but with slightly different error messages. This would make the call stack better traceable...
Let me send you a suggested change to demonstrate what I mean.

Reply:
@sivel this is what I meant: https://github.com/ansible/ansible/pull/80180/files#r1144021641.

Reply:
+1 I'll look over it tomorrow

Reply:
Thanks, it wasn't until I actually applied your recommendation, that I realized it was a completely different place in the code than where I thought you were indicating.